app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -384,6 +384,18 @@ class SectionRenderingService(
       formLevelHeading,
       specialAttributes = Map.empty
     )
+
+    val formModel = formModelOptics.formModelRenderPageOptics.formModel
+    val pageModel = formModel(sectionNumber).allFormComponents
+
+    val fileUploadProviders: List[FileUploadProvider] = pageModel.collect { case IsFileUpload(fu) =>
+      fu.fileUploadProvider
+    }
+
+    val upscanUrl: Option[String] = if (fileUploadProviders.size === 1) {
+      Some(upscanInitiate.get(pageModel.find(_.`type`.isInstanceOf[FileUpload]).get.id).uploadRequest.href)
```

Comment:
This will not work in Add To List, it will find wrong upload  request.

Another issue is that this will fail on `file-upload-frontend` journey. There is missing check on correct provider, ie `FileUploadProvider.Upscan`.